contracts/registries/IPAssetRegistry.sol

```diff
@@ -70,6 +77,43 @@ contract IPAssetRegistry is IIPAssetRegistry, IPAccountRegistry, Governable {
         _metadataProvider = IMetadataProviderMigratable(newMetadataProvider);
     }
 
+    /// @notice Registers an NFT as an IP asset.
+    /// @dev The IP required metadata name and URI are derived from the NFT's metadata.
+    /// @param tokenContract The address of the NFT.
+    /// @param tokenId The token identifier of the NFT.
+    /// @return id The address of the newly registered IP.
+    function register(address tokenContract, uint256 tokenId) external returns (address id) {
+        if (!tokenContract.supportsInterface(type(IERC721).interfaceId)) {
+            revert Errors.IPAssetRegistry__UnsupportedIERC721(tokenContract);
+        }
+
+        if (IERC721(tokenContract).ownerOf(tokenId) == address(0)) {
+            revert Errors.IPAssetRegistry__InvalidToken(tokenContract, tokenId);
+        }
+
+        if (!tokenContract.supportsInterface(type(IERC721Metadata).interfaceId)) {
+            revert Errors.IPAssetRegistry__UnsupportedIERC721Metadata(tokenContract);
+        }
+
+        id = registerIpAccount(block.chainid, tokenContract, tokenId);
+        IIPAccount ipAccount = IIPAccount(payable(id));
+
+        if (bytes(ipAccount.getString("NAME")).length != 0) {
+            revert Errors.IPAssetRegistry__AlreadyRegistered();
+        }
+
+        string memory name = string.concat(IERC721Metadata(tokenContract).name(), " #", tokenId.toString());
```

Comment:
Is there any consideration of multi-chain issue? do you need to add chainid?

Reply:
yes, since the chainid of the NFT, has been encoded into ipaccount address and code, it has already handle the multi-chain issue. but I think it is a good idea to also reflect the chainid in the IP Name. 
Added chainid into the name.